scipy/integrate/_cubature.py

```diff
@@ -207,28 +218,34 @@ def cubature(f, a, b, rule="gk21", rtol=1e-8, atol=0, max_subdivisions=10000,
     # It is also possible to use a custom rule, but this is not yet part of the public
     # API. An example of this can be found in the class scipy.integrate._rules.Rule.
 
-    max_subdivisions = np.inf if max_subdivisions is None else max_subdivisions
+    # If a and b are ordinary Python lists, default to NumPy
+    if isinstance(a, list) and isinstance(b, list):
+        a = np_compat.array(a)
+        b = np_compat.array(b)
+
+    xp = array_namespace(a, b)
```

Comment:
This extra handling seems unnecessary - `array_namespace(a_list)` already returns `np_compat`.